modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -257,6 +258,13 @@ void configureGeneratorProperties() {
             System.out.println(SerializerUtils.toJsonString(openAPI));
         }
 
+        // check to see if we need to apply camelize fix
+        if (config.additionalProperties().containsKey("applyCamelizeFix")) {
```

Comment:
If we think the new method is better, then we should make it enabled by default with an option to opt out and use the legacy method.

Reply:
i want to let users to try it out first to see if we've other edge cases not yet handled.

plan to flip the switch in the upcoming v8.0.0 release.